docs/aws/audit/esmonitoring/rules/es_zone_awareness_enabled.mdx

```diff
@@ -24,6 +24,252 @@ CBP
+
+2. List all Elasticsearch domains: Use the following command to list all the Elasticsearch domains in your AWS account.
+
+   ```
+   aws es list-domain-names
+   ```
+
+   This command will return a list of all Elasticsearch domain names.
+
+3. Describe Elasticsearch domain configuration: For each domain, you need to describe its configuration to check if Zone Awareness is enabled. Use the following command to do this:
+
+   ```
+   aws es describe-elasticsearch-domain-config --domain-name <your-domain-name>
+   ```
+
+   Replace `<your-domain-name>` with the name of your Elasticsearch domain. This command will return the configuration details of the specified Elasticsearch domain.
+
+4. Check Zone Awareness configuration: In the output of the previous command, look for the `ElasticsearchClusterConfig` section. If `ZoneAwarenessEnabled` is `true`, then Zone Awareness is enabled. If it's `false` or not present, then Zone Awareness is not enabled. 
+
+   Here is an example of what the `ElasticsearchClusterConfig` section might look like:
+
+   ```
+   "ElasticsearchClusterConfig": {
+       "InstanceType": "m4.large.elasticsearch",
+       "InstanceCount": 2,
+       "DedicatedMasterEnabled": false,
+       "ZoneAwarenessEnabled": true,
+       "ZoneAwarenessConfig": {
+           "AvailabilityZoneCount": 2
+       },
+       "EBSVolumeType": "gp2",
+       "EBSEnabled": true,
+       "EBSOptions": {
+           "EBSEnabled": true,
+           "VolumeType": "gp2",
+           "VolumeSize": 10
+       }
+   }
+   ```
+
+   In this example, Zone Awareness is enabled because `ZoneAwarenessEnabled` is `true`.
+</Accordion>
+
+<Accordion title='Using Python'>
+To detect whether Elasticsearch has Zone Awareness enabled using Python scripts, you can use the Boto3 library for AWS. Here are the steps:
+
+1. **Import the necessary libraries:**
+
+```python
+import boto3
+from botocore.exceptions import BotoCoreError, ClientError
+```
+
+2. **Create a session using your AWS credentials:**
+
+```python
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='YOUR_REGION_NAME'
+)
+```
+
+3. **Create an Elasticsearch client:**
+
+```python
+es_client = session.client('es')
+```
+
+4. **List all Elasticsearch domains and check if Zone Awareness is enabled:**
+
+```python
+def check_zone_awareness():
+    try:
+        response = es_client.list_domain_names()
+        for domain in response['DomainNames']:
+            domain_name = domain['DomainName']
+            domain_config = es_client.describe_elasticsearch_domain(DomainName=domain_name)
+            domain_zone_awareness = domain_config['DomainStatus']['ElasticsearchClusterConfig']['ZoneAwarenessEnabled']
+            if domain_zone_awareness:
+                print(f"Domain {domain_name} has Zone Awareness enabled.")
+            else:
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>